model/src/main/java/org/projectnessie/model/RefLogResponse.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.model;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import java.util.List;
+import javax.validation.constraints.NotNull;
+import org.eclipse.microprofile.openapi.annotations.enums.SchemaType;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+import org.immutables.value.Value;
+
+@Value.Immutable
+@Schema(type = SchemaType.OBJECT, title = "RefLogResponse")
+@JsonSerialize(as = ImmutableRefLogResponse.class)
+@JsonDeserialize(as = ImmutableRefLogResponse.class)
+public interface RefLogResponse extends PaginatedResponse {
+
+  @NotNull
+  List<RefLogResponseEntry> getLogEntries();
+
+  @Value.Immutable
+  @Schema(type = SchemaType.OBJECT, title = "RefLogResponseEntry")
+  @JsonSerialize(as = ImmutableRefLogResponseEntry.class)
+  @JsonDeserialize(as = ImmutableRefLogResponseEntry.class)
+  interface RefLogResponseEntry {
+    static ImmutableRefLogResponseEntry.Builder builder() {
+      return ImmutableRefLogResponseEntry.builder();
+    }
+
+    String getRefLogId();
+
+    String getRefName();
+
+    String getRefType();
+
+    String getCommitHash();
+
+    List<Ancestor> getAncestors();
+
+    /** time in microseconds. */
+    long getOperationTime();
+
+    String getOperation();
+
+    String getDetails();
```

Comment:
I decided to keep as "Commit on reference $refName with hash $comitHash" in the details and remove commit message. So, if user need more details, they can query commit log api with the information present in the details.

Reply:
That information is available in the respsctive fields - no need to keep another string around.

Reply:
Isn't it better for UI if details has the string? instead of they forming string only for commit operation? Rest of operations also I am providing a nice string. It is duplicate storage for some operations.

Reply:
No, that's actually not worth it. Every UI displays/formats it differently - also l10n i18n

Reply:
@snazy: I think you replied to old message. I don't have details field now.